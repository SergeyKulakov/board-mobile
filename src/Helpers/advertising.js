export const getDataWithAds = data => {
  const result = []

  data.forEach((el, key) => {
    const index = key + 1
    if (data[key + 1]) {
      if (index % 5 === 0) {
        result.push(el)
        result.push('ads')
      } else if (index === 3 || index % 7 === 0) {
        result.push(el)
        result.push('sponsor')
      } else result.push(el)
    } else result.push(el)
  })

  return result
}
