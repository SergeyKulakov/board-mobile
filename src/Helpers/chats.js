import moment from 'moment'

export const getMessageTime = date => {
  const TODAY = moment()
  const YESTERDAY = moment().date(TODAY.get('date') - 1)

  const data = Number(date)

  if (moment(data).isSame(TODAY, 'day')) return moment(data).format('HH:mm')
  if (moment(data).isSame(YESTERDAY, 'day')) return 'Yesterday'
  return moment(data).format('DD MMM YYYY')
}
