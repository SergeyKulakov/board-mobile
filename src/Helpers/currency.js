import data from 'Constants/country-currency-codes'

export const getSearchData = string => {
  if (!string || typeof string !== 'string') return data

  const value = string.toLowerCase()

  return data.filter(
    el =>
      el.country.toLowerCase().indexOf(value) !== -1 ||
      el.country.toLowerCase().indexOf(value) !== -1 ||
      el.code.toLowerCase().indexOf(value) !== -1,
  )
}

export default data
