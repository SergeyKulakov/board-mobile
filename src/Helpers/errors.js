import _ from 'lodash'
import i18n from 'I18N'

export const getError = (error, path = 'apiErrors') => {
  const message = _.get(error, 'payload.message')
  const code = _.get(error, 'payload.code')

  if (
    error.payload === 'Internal server error' ||
    message === 'Internal server error' ||
    message === 'Internal Server Error' ||
    code === 'internal'
  )
    return i18n.t('apiErrors.InternalServerError')

  if (error.message === 'Unauthorized' || error.payload === 'Unauthorized') {
    return i18n.t('apiErrors.unauthorizedError')
  }

  if (!code) return `${message || error.payload}`

  return i18n.t(`${path}.${code}`)
}
