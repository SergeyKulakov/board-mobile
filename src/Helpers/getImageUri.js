import { BACKET } from 'react-native-dotenv'
import ProfileImagesManager from 'Services/Images/ProfileImagesManager'
import { profileImageTypes } from 'Services/Constants/profileImages.constants'
import _ from 'lodash'
import isUrl from 'is-url'

const s3 = 'https://s3.amazonaws.com'
const backet = BACKET
const serviceFolder = 'service-icons'
const jobsFolder = 'job-pics'

export const getServiceImage = fileName => {
  const iconName = fileName.replace(/\s/g, '%20')

  return `${s3}/${backet}/${serviceFolder}/${iconName}`
}

export const getJobImage = (id, fileName) =>
  `${s3}/${backet}/${jobsFolder}/${id}/${fileName}`

export const getProfileImage = ({ type, src, userId }) => {
  let source = src

  if (_.isString(src)) {
    if (/(data:image\/.*?;base64),/g.exec(src) || src.startsWith('http')) {
      source = src
    } else {
      source = ProfileImagesManager.getSignedURLForProfileImage(
        type,
        src,
        userId,
      )
    }
  } else if (_.isObject(src)) {
    source = null
  }

  return source
}

export const getProfileAvatar = (avatarURL, userId) => {
  if (isUrl(avatarURL)) return avatarURL

  if (_.isString(avatarURL)) {
    return getProfileImage({
      type: profileImageTypes.avatar,
      src: avatarURL,
      userId,
    })
  }

  return null
}
