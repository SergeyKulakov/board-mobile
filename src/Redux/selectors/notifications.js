import { createSelector } from 'reselect'

const getReducer = store => store.notifications

export const getNotifications = createSelector(
  getReducer,
  data => data.data,
)
