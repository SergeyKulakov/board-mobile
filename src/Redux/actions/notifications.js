import { createAsyncAction } from 'Helpers/redux'
import { apiCall } from 'Redux/actions/api'
import { secondApi } from 'Constants/api'

export const LOAD_NOTIFICATIONS = createAsyncAction('notifications/LOAD')
export const loadNotifications = callback =>
  apiCall({
    url: secondApi,
    endpoint: '/notification/list',
    types: LOAD_NOTIFICATIONS,
    method: 'GET',
    params: {
      elements_per_page: 20,
    },
    callback,
  })

export const MARK_AS_READ = createAsyncAction('notifications/MARK_READ')
export const markNotificationsAsRead = (notifications = [], callback) =>
  apiCall({
    url: secondApi,
    endpoint: '/notification/mark-as-read',
    types: MARK_AS_READ,
    method: 'POST',
    meta: {
      notifications,
    },
    query: {
      notificationsIds: notifications,
    },
    callback,
  })
