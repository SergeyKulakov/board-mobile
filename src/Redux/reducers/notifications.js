import { createReducer } from 'Helpers/redux'
import { inSocketTypes } from 'Constants/socketEventTypes'
import _ from 'lodash'

import { SIGN_OUT } from 'Redux/actions/auth'
import { LOAD_NOTIFICATIONS, MARK_AS_READ } from 'Redux/actions/notifications'

const initialState = {
  data: [],
  isListEnd: false,
}

const handlers = {
  [LOAD_NOTIFICATIONS.SUCCESS]: (state, { payload }) => ({
    ...state,
    data: _.isArray(payload.list) ? payload.list : state.data,
    isListEnd: payload.isListEnd || state.isListEnd,
  }),
  [`socket/${inSocketTypes.notificationReceived.toUpperCase()}`]: (
    state,
    { payload },
  ) => ({
    ...state,
    data:
      payload.notificationType === 'jobUnavailable'
        ? state.data
        : _.uniqBy([payload, ...state.data], '_id'),
  }),
  [MARK_AS_READ.SUCCESS]: (state, { meta }) => ({
    ...state,
    data: state.data.map(el =>
      meta.notifications.indexOf(el._id) === -1
        ? el
        : {
            ...el,
            read: true,
          },
    ),
  }),
  [SIGN_OUT.REQUEST]: () => initialState,
}

export default createReducer(initialState, handlers)
