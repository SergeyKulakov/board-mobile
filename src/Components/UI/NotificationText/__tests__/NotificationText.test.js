import React from 'react'
import { shallow } from 'enzyme'

import 'jest-styled-components'

import { types } from 'Helpers/notifications'

import Component from '../NotificationText'

describe('MoreText', () => {
  const defaultProps = {
    job: {
      title: 'jobTitle',
      startDate: 1571638857305,
      streetAddress: 'address',
      city: 'city',
      country: 'country',
      zipCode: '123214',
    },
    t: text => text,
    type: undefined,
    user: {
      given_name: 'given_name',
      family_name: 'family_name',
      username: 'username',
    },
    onClickJob: () => null,
    onClickUser: () => null,
  }

  it('render snapshot', () => {
    const defaultWrapper = shallow(<Component {...defaultProps} />)
    const trackingAvailable = shallow(
      <Component {...defaultProps} type={types.trackingAvailable} />,
    )
    const appliedForJob = shallow(
      <Component {...defaultProps} type={types.appliedForJob} />,
    )

    const jobMatchesProfile = shallow(
      <Component {...defaultProps} type={types.jobMatchesProfile} />,
    )

    const newMessage = shallow(
      <Component {...defaultProps} type={types.newMessage} />,
    )

    const newJobRequest = shallow(
      <Component {...defaultProps} type={types.newJobRequest} />,
    )

    const appointmentRemider = shallow(
      <Component {...defaultProps} type={types.appointmentRemider} />,
    )

    const jobRequestRejected = shallow(
      <Component {...defaultProps} type={types.jobRequestRejected} />,
    )

    const jobRequestAccepted = shallow(
      <Component {...defaultProps} type={types.jobRequestAccepted} />,
    )

    const appointmentScheduled = shallow(
      <Component {...defaultProps} type={types.appointmentScheduled} />,
    )

    const jobCancelled = shallow(
      <Component {...defaultProps} type={types.jobCancelled} />,
    )

    const jobUnavailable = shallow(
      <Component {...defaultProps} type={types.jobUnavailable} />,
    )

    expect([
      defaultWrapper,
      trackingAvailable,
      appliedForJob,
      jobMatchesProfile,
      newMessage,
      newJobRequest,
      appointmentRemider,
      jobRequestRejected,
      jobRequestAccepted,
      appointmentScheduled,
      jobCancelled,
      jobUnavailable,
    ]).toMatchSnapshot()
  })
})
