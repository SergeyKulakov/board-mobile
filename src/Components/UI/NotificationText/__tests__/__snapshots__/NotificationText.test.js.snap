// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreText render snapshot 1`] = `
Array [
  <Styled(Text)>
    <Styled(Text) />
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)>
      notifications.trackingJob
       
    </Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
    <Styled(Text)>
       
      notifications.trackingAvailable
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      given_name F.
    </Styled(Text)>
    <Styled(Text)>
       
      notifications.applyJobNotificationMessage
       
    </Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)>
      notifications.followingJob
       
    </Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
    <Styled(Text)>
       
      notifications.mutchesProfile
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      given_name F.
    </Styled(Text)>
    <Styled(Text)>
       
      notifications.sendMessage
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      given_name F.
    </Styled(Text)>
    <Styled(Text)>
       
      notifications.likeToHire
       
    </Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)>
      notifications.appointmentReminder
      : 
    </Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
    <Styled(Text)>
       
      address, city country 123214
       @
       
      October 21, 2019 9:10 AM
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      given_name F.
    </Styled(Text)>
    <Styled(Text)>
       
      notifications.rejectOffer
       
    </Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      given_name F.
    </Styled(Text)>
    <Styled(Text)>
       
      notifications.acceptedOffer
       
    </Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)>
      notifications.NotificationJobBooked
      notifications.NotificationappointmentScheduledFor
       
    </Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
    <Styled(Text)>
      : 
      October 21, 2019 @ 9:10 AM
       - 
      address, city country 123214
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
    <Styled(Text)>
       
      LeftoverOnes.thisJobCancelled
      .
       
      LeftoverOnes.reasonCancelling
      :
    </Styled(Text)>
    <Styled(Text)>
       
    </Styled(Text)>
  </Styled(Text)>,
  <Styled(Text)>
    <Styled(Text)>
      notifications.jobUnavailable
       
    </Styled(Text)>
    <Styled(Text)
      disabled={false}
      onPress={[Function]}
    >
      jobTitle
    </Styled(Text)>
  </Styled(Text)>,
]
`;
